runhouse/resources/hardware/cluster.py

```diff
@@ -1128,6 +1128,24 @@ def _start_or_restart_helper(
         # Update the cluster config on the cluster
         self.save_config_to_cluster()
 
+        # Save the local ~/.rh config to the cluster with the user's hashed token,
+        # if such does not exist on the cluster
+        if rns_client.token:
+            if (
+                self.run(
+                    ["[ -f ~/.rh/config.yaml ]"],
+                    node=self.head_ip,
+                    require_outputs=False,
+                )[0]
+                == 0
+            ):
+                logger.debug("Did not change config.yaml")
+            else:
+                source_path = Path("~/.rh/config.yaml").expanduser()
+                dest_path = "~/.rh/config.yaml"
+                self.rsync(source=source_path, dest=dest_path, up=True, is_file=True)
```

Comment:
this will rsync the whole file, did you look into rsyncing just the portions we want?

Reply:
@jlewitt1 added a comment below